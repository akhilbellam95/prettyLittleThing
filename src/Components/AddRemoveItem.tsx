import { View, Text, StyleSheet, Pressable } from 'react-native';
import React from 'react';

interface AddRemoveItemProps {
  removePressed: () => void;
  addPressed: () => void;
  quantity: string | number;
  style?: object;
}

const AddRemoveItem = ({
  removePressed,
  addPressed,
  quantity,
  style,
}: AddRemoveItemProps): JSX.Element => {
  return (
    <View style={[styles.row, { width: 120, borderColor: 'black', borderWidth: 1, ...style }]}>
      <View style={styles.seperator} />
      <Pressable testID='remove-btn' style={styles.addRemoveBtn} onPress={removePressed}>
        <Text style={styles.textCenter}>-</Text>
      </Pressable>

      <View style={styles.quantityBox}>
        <Text style={styles.quantity}>{quantity}</Text>
      </View>

      <Pressable testID='add-btn' style={styles.addRemoveBtn} onPress={addPressed}>
        <Text style={styles.textCenter}>+</Text>
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  addRemoveBtn: {
    flex: 1,
    backgroundColor: 'yellow',
    justifyContent: 'center',
  },
  quantity: {
    width: 40,
    textAlign: 'center',
  },
  quantityBox: {
    height: 40,
    justifyContent: 'center',
  },
  row: {
    flexDirection: 'row',
  },
  seperator: {
    height: 20,
  },
  textCenter: {
    textAlign: 'center',
  },
});

export default AddRemoveItem;
