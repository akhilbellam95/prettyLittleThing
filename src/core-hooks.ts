import { configureStore, createStore } from "@reduxjs/toolkit";
import { useSelector } from "react-redux";
import { TypedUseSelectorHook, useDispatch } from "react-redux";
import { CartState } from "./Features/Cart/cartSlice";
import { ProductsState } from "./Features/Product/productSlice";

export interface ApplicationState {
  cart: CartState;
  product: ProductsState;
}

export const useAppDispatch = () =>
  useDispatch<ReturnType<typeof configureStore>['dispatch']>();

export const useAppSelector: TypedUseSelectorHook<ApplicationState> =
  useSelector;