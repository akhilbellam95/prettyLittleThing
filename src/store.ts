// eslint-disable-next-line prettier/prettier
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from './Features/Cart/cartSlice';
import productReducer from './Features/Product/productSlice';

export const store = configureStore({
  reducer: {
    cart: cartReducer,
    product: productReducer,
  },
  devTools: process.env.NODE_ENV !== 'production',
});

export type AppDispatch = typeof store.dispatch;

export type RootState = ReturnType<typeof store.getState>;
