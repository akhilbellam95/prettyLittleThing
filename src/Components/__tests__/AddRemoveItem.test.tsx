import { render } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../store';
import AddRemoveItem from '../AddRemoveItem';
import { useAppDispatch, useAppSelector } from '../../core-hooks';

const mockRemovePressed = jest.fn();
const mockAddPressed = jest.fn();

jest.mock('../../core-hooks', () => ({
  useAppSelector: jest.fn(),
  useAppDispatch: jest.fn(),
}));

describe('AddRemoveItem component testing', () => {
  const mockDispatch = jest.fn();
  const mockSelector = jest.fn();

  beforeAll(() => {
    (useAppSelector as jest.Mock).mockImplementation(mockSelector);
    (useAppDispatch as jest.Mock).mockImplementation(mockDispatch);
  });

  it('renders correctly', () => {
    const snap: unknown = render(
      <Provider store={store}>
        <AddRemoveItem
          quantity={30}
          addPressed={mockAddPressed}
          removePressed={mockRemovePressed}
        />
      </Provider>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
