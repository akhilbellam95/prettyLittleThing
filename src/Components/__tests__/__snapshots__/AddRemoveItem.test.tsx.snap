// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRemoveItem component testing renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "borderColor": "black",
        "borderWidth": 1,
        "width": 120,
      },
    ]
  }
>
  <View
    style={
      {
        "height": 20,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "yellow",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="remove-btn"
  >
    <Text
      style={
        {
          "textAlign": "center",
        }
      }
    >
      -
    </Text>
  </View>
  <View
    style={
      {
        "height": 40,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "textAlign": "center",
          "width": 40,
        }
      }
    >
      30
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "yellow",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="add-btn"
  >
    <Text
      style={
        {
          "textAlign": "center",
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;
